services/src/main/java/org/apache/druid/cli/CliPeon.java

```diff
@@ -301,7 +302,7 @@ public String getTaskIDFromTask(final Task task)
   public void run()
   {
     try {
-      Injector injector = makeInjector();
+      Injector injector = makeInjector(ImmutableSet.of(NodeRole.PEON));
```

Comment:
If you are adding this here, you should also remove the binding created lower down in the file.

Reply:
done, thanks!